content/engine/security/rootless.md

```diff
@@ -58,6 +58,41 @@ testuser:231072:65536
   ([Ubuntu-specific kernel patch](https://kernel.ubuntu.com/git/ubuntu/ubuntu-bionic.git/commit/fs/overlayfs?id=3b7da90f28fe1ed4b79ef2d994c81efbc58f1144)).
 
 - Known to work on Ubuntu 18.04, 20.04, and 22.04.
+
+- Ubuntu 24.04 and later enables restricted unprivileged user namespaces by
+  default, which prevents unprivileged processes in creating user namespaces
+  unless an AppArmor profile is configured to allow programs to use
+  unprivileged user namespaces. If you install the rootless extras using the
+  [installation script](https://get.docker.com/rootless), you must add an
+  AppArmor profile for `rootlesskit` as follows:
+
+  ```bash
+  if [ ! -e "/etc/apparmor.d/usr.local.bin.rootlesskit" ] && [ -e "/etc/apparmor.d/abi/4.0" ] && [ -e "/proc/sys/kernel/apparmor_restrict_unprivileged_userns" ]; then
+    cat >"/etc/apparmor.d/usr.local.bin.rootlesskit" <<EOF
+  # Ubuntu 23.10 introduced kernel.apparmor_restrict_unprivileged_userns
+  # to restrict unsharing user namespaces:
+  # https://ubuntu.com/blog/ubuntu-23-10-restricted-unprivileged-user-namespaces
+  #
+  # kernel.apparmor_restrict_unprivileged_userns is still opt-in in Ubuntu 23.10,
+  # but it is expected to be enabled in future releases of Ubuntu.
+  abi <abi/4.0>,
+  include <tunables/global>
+
+  /usr/local/bin/rootlesskit flags=(unconfined) {
```

Comment:
The path is /home/USERNAME/bin/rootlesskit in the case of https://get.docker.com/rootless

Reply:
thank you, I've updated to add the config discretely, with the heredoc containing the $HOME variable. Hope it's ok